components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -1783,134 +1783,362 @@ private int handleStartIndexEqualsNULL(Integer startIndex) {
-                    "attribute value : " + attributeValue + " and filter operation " + filterOperation, e);
+            throw new CharonException(
+                    "Error in filtering groups by attribute name : " + attributeName + ", " + "attribute value : "
+                            + attributeValue + " and filter operation " + filterOperation, e);
         } catch (org.wso2.carbon.user.api.UserStoreException e) {
-            throw new CharonException("Error in filtering group with filter: "
-                    + attributeName + filterOperation + attributeValue, e);
-        } catch (IdentitySCIMException e) {
-            throw new CharonException("Error in retrieving SCIM Group information from database.", e);
-        } catch (BadRequestException e) {
-            throw new CharonException("Error in retrieving SCIM Group.", e);
+            throw new CharonException(
+                    "Error in filtering group with filter: " + attributeName + filterOperation + attributeValue, e);
         }
-        //set the totalResults value in index 0
+        // Set the totalResults value in index 0.
         filteredGroups.set(0, filteredGroups.size() - 1);
         return filteredGroups;
     }
 
+    private void getRolesWithAttributes(String roleName, List<Object> filteredGroups)
+            throws CharonException, UserStoreException {
+
+        String userStoreDomainName = IdentityUtil.extractDomainFromName(roleName);
+        if (isInternalOrApplicationGroup(userStoreDomainName) || isSCIMEnabled(userStoreDomainName)) {
+            if (log.isDebugEnabled()) {
+                log.debug("SCIM is enabled for the user-store domain : " + userStoreDomainName + ". "
+                        + "Including group with name : " + roleName + " in the response.");
+            }
+            try {
+                Group group = getGroupWithName(roleName);
+                filteredGroups.add(group);
+            } catch (IdentitySCIMException e) {
+                throw new CharonException("Error in retrieving SCIM Group information from database.", e);
+            } catch (BadRequestException e) {
+                throw new CharonException("Error in retrieving SCIM Group.", e);
+            }
+        } else {
+            if (log.isDebugEnabled()) {
+                log.debug("SCIM is disabled for the user-store domain : " + userStoreDomainName + ". Hence "
+                        + "group with name : " + roleName + " is excluded in the response.");
+            }
+        }
+    }
+
+    /**
+     * Method to append primary domain name for the roles in the primary user store.
+     *
+     * @param roleName Role name
+     * @return Updated role name for roles in primary domain. Return same value for roles in other domains.
+     */
+    private String appendPrimaryDomainNameForRoleName(String roleName) {
+
+        //construct the group name with domain -if not already provided, in order to support multiple user
```

Comment:
Follow the comment convention, start with a capital and ends with full stop.

Reply:
Fixed: e0b27781f03c5dc319612f3d1924e70dce6c2f21